lib/elixir/lib/keyword.ex

```diff
@@ -1316,4 +1316,43 @@ defmodule Keyword do
   def size(keywords) do
     length(keywords)
   end
+
+  @doc """
+  Filters `keywords` i.e. returns a keyword list containing
+  only the elements from `keywords` for which the function
+  `fun` returns a truthy value.
+
+  See also `reject/2` which discards all elements where the function
+  returns a truthy value.
+
+  ## Examples
+
+      iex> Keyword.filter([one: 1, two: 2, three: 3], fn {_key, val} -> rem(val, 2) == 1 end)
+      [one: 1, three: 3]
+
+  """
+  @doc since: "1.13.0"
+  @spec filter(t, ({key, value} -> as_boolean(term))) :: t
+  def filter(keywords, fun) when is_list(keywords) and is_function(fun, 1) do
+    :lists.filter(keywords, fun)
```

Comment:
Ah, we have to implement this in Elixir because Erlang doesn’t have our concept of truthiness.

Reply:
Even with `inline_list_funcs` it seems like the Erlang compiler is not able to currently optimize intermediate closures, so the final implementation is a hand-written tail-recursive implementation.